src/encoded/static/components/rnaget.js

```diff
@@ -657,42 +694,81 @@ class RNASeqMatrixSearch extends TextFilter {
     constructor(props) {
         super();
 
-        this.handleChange = this.handleChange.bind(this);
+        this.onUnitsChange = this.onUnitsChange.bind(this);
+        this.onKeyDown = this.onKeyDown.bind(this);
         this.handleInputChange = this.handleInputChange.bind(this);
         this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleAutocompleteClick = this.handleAutocompleteClick.bind(this);
+        this.lastGene = this.lastGene.bind(this);
 
         this.state = {
             unitsOption: props.query.getKeyValuesIfPresent('units').join(','),
             genes: props.query.getKeyValuesIfPresent('genes').join(','),
+            showAutoSuggest: false,
         };
     }
 
-    handleChange(e) {
+    onUnitsChange(e) {
         this.setState({ unitsOption: e.target.value });
     }
 
-    handleInputChange(e) {
-        this.setState({ genes: e.target.value });
+    lastGene() {
+        const inputNode = this.geneInput;
+
+        if (inputNode === undefined) {
+            return '';
+        }
+
+        const geneTerms = inputNode.value.split(',');
+        return (geneTerms[geneTerms.length - 1].trim());
+    }
+
+    handleInputChange() {
+        this.setState({ showAutoSuggest: (this.lastGene() !== ''), genes: this.geneInput.value });
+    }
+
+    onKeyDown() {
+        this.setState({ showAutoSuggest: (this.lastGene() !== '') });
     }
 
     handleSubmit(e) {
         e.preventDefault();
         window.location.href = `/rnaget?genes=${this.state.genes}&units=${this.state.unitsOption}`;
     }
 
+    handleAutocompleteClick(term) {
+        const inputNode = this.geneInput;
+
+        let geneTerms = inputNode.value.split(',');
```

Comment:
This looks hacky. You declare a variable and then change it with itself...

Reply:
this is mostly for readability, I can change it if it breaks a JS pattern it is used